Api keys page

werronense opened the pull request:
Adds API key table and modal to API key page.

werronense committed:
Create API key table

Signed-off-by: Stephen Werronen <stephen.werronen@gmail.com>

werronense committed:
Add modal and mock key functions

Signed-off-by: Stephen Werronen <stephen.werronen@gmail.com>

YourTechBud reviewed (changes requested):
I think we should move all the components in a new `api-key` directory in the `src/ui/widgets` directory.

werronense committed:
Fix component architecture

Signed-off-by: Stephen Werronen <stephen.werronen@gmail.com>

werronense commented:
@YourTechBud I've refactored the component structure and fixed the compiler errors from the linter. It should be able to build and deploy now. Please review and approve.
